src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Key.java

```diff
@@ -487,6 +497,41 @@ private static abstract class P11PublicKey extends P11Key implements
         }
     }
 
+    static final class P11PBEKey extends P11SecretKey
+            implements PBEKey {
+        private static final long serialVersionUID = 6847576994253634876L;
+        private final char[] password;
+        private final byte[] salt;
+        private final int iterationCount;
+        P11PBEKey(Session session, long keyID, String algorithm,
+                int keyLength, CK_ATTRIBUTE[] attributes,
+                char[] password, byte[] salt, int iterationCount) {
+            super(session, keyID, algorithm, keyLength, attributes);
+            this.password = password.clone();
+            this.salt = salt.clone();
+            this.iterationCount = iterationCount;
+        }
+
+        @Override
+        public char[] getPassword() {
+            return password.clone();
```

Comment:
Would you consider throw IllegalStateException if password is cleared? This would help flag the wrong usage case of calling getPassword() after the password has been cleared.

Reply:
Sounds good